File: M/1954-minimum-garden-perimeter-to-collect-enough-apples/main.go
package main

import (
	"fmt"
)

func minimumPerimeter(neededApples int64) int64 {
	var sum int64
	sum = 0
	var i int64
	i = 0
	for sum < neededApples {
		i++
		sum += 4*(i*(i+1)+i*(2*i+1)) - 4*2*i
	}

	return i * 8
}

func main() {
	fmt.Println()
}

File: M/1954-minimum-garden-perimeter-to-collect-enough-apples/main.rs
impl Solution {
  pub fn minimum_perimeter(needed_apples: i64) -> i64 {
    let mut sum: i64 = 0;
    let mut i: i64 = 0;
    while sum < needed_apples {
      i += 1;
      sum += 4 * (i * (i + 1) + i * (2 * i + 1)) - 4 * 2 * i;
    }

    i * 8
  }
}
